### ISSUE TITLE
cls_decoder doesn't exist in pretrained weight

### ISSUE THREAD

**Issue by taliq** [bowang-lab/scGPT#78]
Hi, thanks for the effort to maintain the repo.
I found the uploaded pre-trained weight doesn't have the cls_decoder which seems important for cell embedding.
I assume this version is trained before modifying the model, do you have a plan to share the recent weights?

**Comment by subercui** [comment]
Hi, thank you for the question. I think there may be a slight misunderstanding.

The cls_decoder uses the cell embedding at the position of the <cls> token to perform tasks like cell type annotation. It is only used when the model needs to output the cell type classification predictions. You may find it here https://github.com/bowang-lab/scGPT/blob/0d08d611ece2c69a9e74ffdfd5c6a90986b4726c/scgpt/model/model.py#L364 and description in the function docstring. 

So you don't necessarily need the cls_decoder to train the cell embedding. We didn't use any cell type classification loss during pretraining, and we also did not include that decoder in the pretraining. That explains why it is not in the shared checkpoint files. In any of the fine-tuning tutorials, including in the cell type annotation notebook where the cls_decoder is used, this should be fine because we loaded the model weights by matching the parameter names.

To further clarify, the model weights we shared are the correct versions in this sense. Hope these explanations make sense to you
